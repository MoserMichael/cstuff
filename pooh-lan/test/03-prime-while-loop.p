

# get all prime numbers up to max
sub primes( max )

  mark = []
  ret = []

  n = 2 
  while n < max
    m = 2 * n
    loop 
        if m > max
            break;
        end
        mark[ m ] = 1
        m = m + n
    end
    n = n + 1
  end

  n = 2
  while n  < max
    if mark[ n ] != 1
       push( ~array ret ~top n )
    end
    n = n + 1
  end

  return ret
end


# get prime numbers up to 100
pr = primes( ~max 100 )

# print them out.
i = 1
while i <= size( ~arg pr)
# println( ~msg 'prime number [ pr[i] ]' )
  println( ~msg 'prime number ' ..  pr[i]  )

  i = i + 1
end
