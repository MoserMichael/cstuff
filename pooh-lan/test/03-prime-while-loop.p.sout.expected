prime number 2
prime number 3
prime number 5
prime number 7
prime number 11
prime number 13
prime number 17
prime number 19
prime number 23
prime number 29
prime number 31
prime number 37
prime number 41
prime number 43
prime number 47
prime number 53
prime number 59
prime number 61
prime number 67
prime number 71
prime number 73
prime number 79
prime number 83
prime number 89
prime number 97
